pkg/orchestrator/provider/aws/scanestimation/estimate.go

```diff
@@ -78,17 +78,84 @@ const (
 	DataTransfer       recipeResource = "DataTransfer"
 )
 
+// TODO add infoFinder family stats.
+var families = []familiestypes.FamilyType{
+	familiestypes.SBOM, familiestypes.Vulnerabilities, familiestypes.Secrets,
+	familiestypes.Exploits, familiestypes.Rootkits, familiestypes.Misconfiguration, familiestypes.Malware,
+}
+
 // Static times from lab tests of family scan duration in seconds per GB.
 // The tests were made on a t2.large instance with a gp2 volume.
+// The times correspond to the scan size values {0.01, 2.5, 8.1}.
 // nolint:gomnd
-var familyScanDurationPerGBMap = map[familiestypes.FamilyType]time.Duration{
-	familiestypes.SBOM:             time.Duration(4.5 * float64(time.Second)),
-	familiestypes.Vulnerabilities:  1 * time.Second, // TODO check time with no sbom scan
-	familiestypes.Secrets:          300 * time.Second,
-	familiestypes.Exploits:         0,
-	familiestypes.Rootkits:         0,
-	familiestypes.Misconfiguration: 1 * time.Second,
-	familiestypes.Malware:          360 * time.Second,
+var familyScanDurationsMap = map[familiestypes.FamilyType][]float64{
+	familiestypes.SBOM:             {0.01, 11, 37},
+	familiestypes.Vulnerabilities:  {0.01, 1, 11}, // TODO check time with no sbom scan
+	familiestypes.Secrets:          {0.01, 720, 1320},
+	familiestypes.Exploits:         {0, 0, 0},
+	familiestypes.Rootkits:         {0, 0, 0},
+	familiestypes.Misconfiguration: {0.01, 4, 5},
+	familiestypes.Malware:          {0.01, 840, 2460},
+}
+
+// familyCoefficients populated during init in order to extrapolate the scan duration from scan size per family.
+// The extrapolation formula is y = a * ln(x) + b.
+var familyCoefficients = make(map[familiestypes.FamilyType][]float64, len(families))
```

Comment:
If each coefficient is a known set of two float64s, I would much rather we use a struct here.

```
// struct representing a model for a specific family
type familyCofficient struct {
   a float64
   b float64
}
```

we could even add the functions on the struct:

```
func (fc *familyCofficient) calculateDuration(sizeInGb float64) (float64) {
     return fc.a * sizeInGb + fc.b
}
```

Reply:
agree